### PR TITLE
Add GitHub OAuth

### PR DESCRIPTION
**Pull request by Ellahinator**
GitHub OAuth. Follow this guide: [Login with GitHub](https://supabase.com/docs/guides/auth/social-login/auth-github?queryGroups=environment&environment=server&queryGroups=framework&framework=nextjs)

image

image

### CODE REVIEW

File: app/(auth)/actions.ts [hunk 1]
```diff
@@ -51,15 +51,15 @@ export async function signup(formData: FormData) {
   redirect("/");
 }
 
-// Google OAuth sign-in
-export async function signinWithGoogle() {
-
+// OAuth sign-in with Google or GitHub
+export async function signinWithOAuth(provider: 'google' | 'github') {
```
Comment (Fryingpannn) [inline app/(auth)/actions.ts:55]: Let's make an enum for providers

E.g.
```
enum Colors {
  RED = 'RED',
  GREEN = 'GREEN',
  BLUE = 'BLUE'
}

// Usage
const myColor: Colors = Colors.RED;
```

**Comment by Fryingpannn** [comment]
Noice

File: app/(auth)/actions.ts (hunk 1 above)
Comment (Ellahinator) [reply app/(auth)/actions.ts]: Now using built-in supabase `Provider` type: https://github.com/trypear/pear-landing-page/pull/38/commits/9473ed1d363e085e07f262fff4b482d2e1ab4e66

**Comment by Ellahinator** [comment]
@Fryingpannn this is up to date and should be good to merge preferably before the UI changes of the auth pages get pushed.

**Comment by Ellahinator** [comment]
Will fix merge conflicts

**Comment by Ellahinator** [comment]
Moving these changes to
- https://github.com/trypear/pear-landing-page/pull/42

**Pull request closed without merging**